modules/login/src/main/resources/META-INF/resources/login/login.jsp

```diff
@@ -1,199 +1,375 @@
+                        <p>Please fill in the required account information.</p>
+
+                    </div>
+                    <div class="modal-body">
+                        <div style="display: table;margin: auto;width: 90%">
+
+
+                            <div class="col-sm-6">
+                                <div class="form-group">
+                                    <div class="input-group" style="width:100%">
+                                        <span class="input-group-addon glyphicon glyphicon-user" style="top: 0"></span>
+                                        <label for="firstname" class="sr-only">First Name</label>
+                                        <input type="text" id="firstname" class="form-control requiredtext"
+                                               placeholder="First Name" required>
+                                    </div>
+                                </div>
+                            </div>
+                            <div class="col-sm-6">
+                                <div class="form-group">
+                                    <div class="input-group" style="width:100%">
+                                        <span class="input-group-addon glyphicon glyphicon-user" style="top: 0"></span>
+                                        <input type="text" id="lastname"
+                                               class="form-control requiredtext"
+                                               placeholder="Enter Last Name" required></div>
+                                </div>
+                            </div>
+                            <div class="col-xs-12">
+                                <div class="form-group">
+                                    <div class="input-group" style="width:100%">
+                                        <span class="input-group-addon glyphicon glyphicon-envelope" style="top: 0"></span>
+                                        <input type="email" id="regemail" name="email"
+                                               class="form-control requiredtext"
+                                               placeholder="Work email"
+                                               data-pattern-error="Invalid Email Address"
+                                               data-remote-error="The email is already registered"
+                                               pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,63}$"
+                                               data-remote="http://nicsweb:8082/nics/verifyData" required>
+
+                                    </div>
+                                    <div class="help-block with-errors"></div>
+                                </div>
+                            </div>
+                            <div class="col-sm-6">
+                                <!-- New pattern: pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@#$%_!-])[a-zA-Z0-9@#$%_!-]{8,20}$" -->
```

Comment:
Could take these out. I found it useful since it took a while to realize that the front-end pattern had to match somewhere on the backend.